dtfs-central-api/src/v1/controllers/utilisation-report-service/post-keying-data.controller/helpers.ts

```diff
@@ -0,0 +1,34 @@
+import { FeeRecordEntity } from '@ukef/dtfs2-common';
+import { FeeRecordRepo } from '../../../../repositories/fee-record-repo';
+
+export type GenerateKeyingDataDetails = {
+  feeRecord: FeeRecordEntity;
+  generateKeyingData: boolean;
+}[];
+
+const getFeeRecordFacilityIdsWhichCannotBeKeyed = async (): Promise<Set<string>> => {
+  const feeRecordsWhichCannotBeKeyed = await FeeRecordRepo.findByStatuses(['TO_DO', 'DOES_NOT_MATCH']);
+  return feeRecordsWhichCannotBeKeyed.reduce((facilityIds, { facilityId }) => facilityIds.add(facilityId), new Set<string>());
```

Comment:
I think this was just an oversight as I was doing this - we should only be checking fee records on the actual report as (at least, in theory), there could exist a fee record for a different report with the same facility id which has not been moved to `MATCH` yet, which would cause this to behave incorrectly